### PR TITLE
spacewalk-proxy: remove unnecessary dependencies to Squid

### PR DESCRIPTION
**Pull request by moio**
## What does this PR change?

Removes unnecessary dependencies between packages, this is useful for the ongoing containerization effort.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by moio** [comment]
@mcalmer I could not see a good reason to keep the requirement in e337d2d32537ddd2c2c46b16c005c99b837dc5d1 - tried out building a branch of this package from my home project and installation/operation works as expected.

Do you see something I might have missed about it?

**Comment by mcalmer** [comment]
@moio take care, that squid is in our proxy pattern. In case we install the product on a real system, squid is needed to be installed.

**Comment by moio** [comment]
@mcalmer it is not, but `squid` is still pulled in by `spacewalk-proxy-management`. Is that OK or we want an explicit mention in the pattern?

**Comment by mcalmer** [comment]
@moio Ok, this will work.

**Review by mcalmer — APPROVED** [review]
Maybe you need a changelog. This package is not touched so often and to submit it we need at least 1 changelog entry.

**Comment by juliogonzalez** [comment]
@moio yes, a changelog entry is needed.

**Comment by moio** [comment]
PR test results (no difference from `master` at the time of writing): https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/894/testReport/

File: proxy/proxy/spacewalk-proxy.changes [hunk 1]
```diff
@@ -1,3 +1,8 @@
+-------------------------------------------------------------------
+Fri Dec 10 17:54:36 UTC 2021 - Silvio Moioli <moio@suse.com>
+
```
Comment (mcalmer) [inline proxy/proxy/spacewalk-proxy.changes:3]: Wrong format. Please remove the header

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: proxy/proxy/spacewalk-proxy.changes (hunk 1 above)
Comment (moio) [reply proxy/proxy/spacewalk-proxy.changes]: Sorry about that, fixed

**Review by mcalmer — APPROVED** [review]

**Pull request merged**